src/core/connection-parameters.cpp

```diff
@@ -240,23 +240,43 @@ getPossiblyQuotedWord(std::string const &s, std::string::const_iterator &i)
     std::string::const_iterator const end = s.end();
     skipWhiteSpace(i, end);
 
+    if (i == end)
+    {
+        return {};
+    }
+
+    char quote = '"';
+    if (*i == '\'')
+    {
+        quote = '\'';
+    }
+
     std::string word;
 
-    if (i != end && *i == '"')
+    if (i != end && *i == quote)
     {
         for (;;)
         {
             if (++i == end)
             {
                 std::ostringstream os;
-                os << "Expected '\"' not found before the end of the string "
+                os << "Expected ";
+                if (quote == '\'')
+                {
+                    os << "'";
```

Comment:
If you want to show it differently I'd do this for symmetry:
```suggestion
                    os << R"("'")";
```

but I think we could also just use the `else` version in all cases.